src/main/java/strategy/FruitStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package strategy;
+
+import java.util.Map;
+import model.Fruit;
+import operation.Operation;
+
+public class FruitStrategyImpl implements FruitStrategy {
+    private Map<Fruit.Operation, Operation> operationOperationHandlerMap;
+
+    public FruitStrategyImpl(Map<Fruit.Operation, Operation> operationOperationHandlerMap) {
+        this.operationOperationHandlerMap = operationOperationHandlerMap;
+    }
+
+    @Override
+    public Operation get(Fruit.Operation operation) {
+        return operationOperationHandlerMap.get(operation);
+    }
```

Comment:
```suggestion
    @Override
    public void proceed(Fruit fruit) {
        return operationOperationHandlerMap.get(fruit.getOperation()).proceed(fruit);
    }
```

lets move this logic to this method